regionServiceClient/lib/cloudregister/registerutils.py

```diff
@@ -0,0 +1,419 @@
+def import_smtcert_12(smt):
+    """Import the SMT certificate on SLES 12"""
+    key_chain = '/usr/share/pki/trust/anchors'
+    if not smt.write_cert(key_chain):
+        return 0
+    if not update_ca_chain(['update-ca-certificates']):
+        return 0
+
+    return 1
+
+
+# ----------------------------------------------------------------------------
+def import_smt_cert(smt):
+    """Import the SMT certificate for the given server"""
+    import_result = None
+    if is_sles11():
+        import_result = import_smtcert_11(smt)
+    else:
+        import_result = import_smtcert_12(smt)
+    if not import_result:
+        logging.error('SMT certificate import failed')
+        return None
+
+    return 1
+
+
+# ----------------------------------------------------------------------------
+def is_registered(smt):
+    """Firgure out if any of the servers is known and this instan"""
+    if check_registration(smt.get_FQDN()):
+        return 1
+
+    return None
+
+
+# ----------------------------------------------------------------------------
+def is_sles11():
+    """Return true if this is SLES 11"""
+    if os.path.exists('/etc/SuSE-release'):
+        content = open('/etc/SuSE-release', 'r').readlines()
+        for ln in content:
+            if 'SUSE Linux Enterprise Server 11' in ln:
+                return True
+
+    return False
+
+
+# ----------------------------------------------------------------------------
+def set_as_current_smt(smt):
+    """Store the given SMT as the current SMT server."""
+    if not os.path.exists(REGISTRATION_DATA_DIR):
+        os.system('mkdir -p %s' % REGISTRATION_DATA_DIR)
+    store_smt_data(__get_registered_smt_file_path(), smt)
+
+
+# ----------------------------------------------------------------------------
+def remove_registration_data(smt_servers):
+    """Reset the instance to an unregistered state"""
+    smt_data_file = __get_registered_smt_file_path()
+    if os.path.exists(smt_data_file):
+        os.unlink(smt_data_file)
+    for smt in smt_servers:
+        server_name = smt.get_FQDN()
+        domain_name = smt.get_domain_name()
+        clean_hosts_file(domain_name)
+        __remove_credentials(server_name)
+        __remove_repos(server_name)
+        __remove_service(server_name)
+
+
+# ----------------------------------------------------------------------------
+def replace_hosts_entry(current_smt, new_smt):
+    """Replace the information of the SMT server in /etc/hosts"""
+    known_hosts = open('/etc/hosts', 'r').readlines()
+    new_hosts = ''
+    current_smt_ip = current_smt.get_ip()
+    for entry in known_hosts:
+        if current_smt_ip in entry:
+            new_hosts += '%s\t%s\t%s\n' % (new_smt.get_ip(),
+                                           new_smt.get_FQDN(),
+                                           new_smt.get_name())
+            continue
+        new_hosts += entry
+
+    hosts = open('/etc/hosts', 'w')
```

Comment:
Then I need to do a conversion of data type, in clean_hosts_file the content is treated in list context and here as a string. The conversion takes as many lines as it takes to keep tw implementation of the writing, vs. write_hosts_file(content) thus from my perspective it is equivalent to the current implementation.